models/demos/resnet/tt/metalResnetBlock50.py

```diff
@@ -1401,40 +1424,59 @@ def __init__(
             ] = hardcoded_conv_blocking_and_parallelization_config[batch_size][
                 (first_conv_output_padded_nhw_size, first_conv_output_channels)
             ]
-            sliding_window_op_params = [
-                (1, 1),
-                (0, 0),
-                (4, 4),
-                (batch_size, 115, 115),
-                grid_size,
-                self.first_conv_num_cores_nhw,
-            ]
+
+            sliding_window_op_params = SlidingWindowOpParamsWithParallelConfig(
+                stride_h=1,
+                stride_w=1,
+                pad_h=0,
+                pad_w=0,
+                window_h=4,
+                window_w=4,
+                batch_size=batch_size,
+                input_h=115,
+                input_w=115,
+                num_cores_h=grid_size[1],
+                num_cores_w=grid_size[0],
+                num_cores_nhw=self.first_conv_num_cores_nhw,
+            )
+
             per_core_act_h_ntiles = (int)(per_core_act_h / 32)
             per_core_weight_w_ntiles = (int)(per_core_weight_w / 32)
             conv1_weight = pad_and_fold_conv_filters_for_unity_stride(conv1_weight, 2, 2)
-            self.conv1 = TTPyConv(
+            assert act_block_w_datums % 32 == 0
+            assert act_block_h_datums % 32 == 0
+            assert weight_block_w_datums % 32 == 0
+            assert out_subblock_h_datums % 32 == 0
+            assert out_subblock_w_datums % 32 == 0
+            assert out_block_h_datums % 32 == 0
+            config_override = {
+                "act_block_w_ntiles": act_block_w_datums // 32,
```

Comment:
@tt-nshanker maybe we could have this as  `act_block_w` instead of `act_block_w_ntiles` and then assert and division can be done internally in the OP.
Tests wouldn't have to replicate these checks / less code, and cleaner?